### PR TITLE
Added: Navigation and Routing

### LINKED ISSUE #2263: Welcome - Navigation and Routing
Description

Add a "Home" page to the navigation bar, represented by the Home icon.
Integrate the new "Statistics" and "Cases" routes into the application's routing structure within the authenticated layout.
Modify authentication redirects to point to the "Statistics" page.
Code References (Example):

```
 // useNavbarLogic.tsx
 import { Home } from 'lucide-react';

 const navItems = [ 
     {
         label: 'Home',
         href: '/en/statistics',
         icon: Home 
     }, 
     // ...existing navItems
 ]; 

// Router.tsx

{
        element: <AuthenticatedLayout />,
        loader: authenticatedLayoutLoader,
        errorElement: <RouteError />,
        children: [
          {
            path: '/:locale',
            element: <Locale />,
            errorElement: <RouteError />,
            children: [
                // ... existing children
               {
                path: '/:locale/statistics',
                element: <Statistics />,
                errorElement: <RouteError />,
               },
              {
                path: '/:locale/workflows',
                element: <Workflows />,
                errorElement: <RouteError />,
               }
            ]
           }
        ]
}

```

```
// Providers.tsx

      <AuthProvider
        redirectAuthenticatedTo={'/en/statistics'}
        signInOptions={{
          callbackUrl: state?.from
            ? ...
            : '/en/statistics',
        }}
       // ...rest of props
      >
```

### PR DESCRIPTION
**Pull request by shashankvish0010**
### Description
Added a "Home" page to the navigation bar, represented by the 'Home' icon.
Integrated the "Statistics" page and "Workflows" page routes into the application's routing structure within the authenticated layout for now.
I modified the authentication redirect; now it redirects to statistics.

fixes: #2263 

### How these changes were tested
 Tested with local development setup.

### Examples and references
Screenshot (319)

### CODE REVIEW

**Comment by Omri-Levy** [comment]
Great work! Before merging try and see if you can understand why the nav item is not styled as the active item. Notice its a shade of gray instead of the expected black.

**Comment by shashankvish0010** [comment]
> Great work! Before merging try and see if you can understand why the nav item is not styled as the active item. Notice its a shade of gray instead of the expected black.

Thanks for addressing the issue, I checked it and got to know that the 'Home' nav-item is not styled as active item when it's active, and added changes to make sure it works perfectly fine.

File: apps/backoffice-v2/src/Router/Router.tsx [hunk 1]
```diff
@@ -46,12 +48,12 @@ const router = createBrowserRouter([
...
-                      element: <SignIn />,
-                      errorElement: <RouteError />,
-                    },
-                  ]
+                  {
+                    path: '/:locale/auth/sign-in',
+                    element: <SignIn />,
+                    errorElement: <RouteError />,
+                  },
+                ]
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/Router/Router.tsx:56]: Please ensure your IDE is configured to run Prettier and ESLint on save. Install the respective plugins if you haven't and make sure you're not running a custom format on save. 🙏🏼

File: apps/backoffice-v2/src/Router/types.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ export type TRouteWithoutChildren = {
 
 export type TRouteWithChildren = {
   text: ReactNode | ReactNode[];
+  href?: string;
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/Router/types.ts]: Revert this change. Routes with children will never have an href. Also, having an optional href will leave room for mistakes.

File: apps/backoffice-v2/src/common/components/organisms/Header/Header.Navbar.tsx [hunk 3]
```diff
@@ -86,7 +86,7 @@ export const Navbar: FunctionComponent = () => {
                   className={ctw(
                     `flex items-center gap-x-1 px-1.5 py-1 text-sm font-semibold capitalize text-[#8990AC] hover:bg-[#EBEEF9] hover:text-[#5E688E] active:bg-[#e0e4f6]`,
                     {
-                      'bg-white text-[#20232E]': navItem.filterId === filterId,
+                      'bg-white text-[#20232E]': navItem.filterId === filterId || (navItem.key === 'nav-item-Home' && navItem.href === window.location.pathname),
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/common/components/organisms/Header/Header.Navbar.tsx]: The fix should work for any item without children and not use the item's key. Note since we're in React and using React router here you'll use the pathname from `useLocation` and not from the window directly. 🙂

File: apps/backoffice-v2/src/common/components/organisms/Header/hooks/useNavbarLogic/useNavbarLogic.tsx [hunk 4]
```diff
@@ -57,9 +63,13 @@ export const useNavbarLogic = () => {
...
-      return navItem.children?.some(
-        childNavItem => childNavItem.filterId === filterId || childNavItem.href === pathname,
-      );
+      if (navItem.children) {
+        return navItem.children?.some(
+          childNavItem => childNavItem.filterId === filterId || childNavItem.href === pathname,
+        );
+      } else {
+        return navItem.href === pathname;
+      }
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/common/components/organisms/Header/hooks/useNavbarLogic/useNavbarLogic.tsx]: Let's revert this considering the other suggested changes. 🙂  Also consider the following in the future for readability by early returning on the simpler condition (usually the negative).
```suggestion
     if (!navItem.children) {
        return navItem.href === pathname;
     }

     return navItem.children?.some(
       childNavItem => childNavItem.filterId === filterId || childNavItem.href === pathname,
     );
```

File: apps/backoffice-v2/src/common/components/organisms/Header/Header.Navbar.tsx [hunk 5]
```diff
@@ -86,7 +86,7 @@ export const Navbar: FunctionComponent = () => {
...
                     },
                   )}
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/common/components/organisms/Header/Header.Navbar.tsx]:
```suggestion
                  className={({ isActive }) =>
                    ctw(
                      `flex items-center gap-x-1 p-2 text-sm font-semibold capitalize hover:bg-[#EBEEF9] hover:text-[#5E688E] active:bg-[#e0e4f6] [&:not([aria-current=page])]:text-[#8990AC]`,
                      {
                        'bg-white text-[#20232E]': isActive,
                      },
                    )
                  }
```

Comment (Omri-Levy) [reply apps/backoffice-v2/src/common/components/organisms/Header/Header.Navbar.tsx]: Please remove `className` on `TNavItemProps` to allow this change and update the `NavItem` component with the code below to allow this change. Thanks!
```
        className={props =>
          ctw(
            `flex gap-x-2 rounded-md`,
            className instanceof Function ? className(props) : className,
          )
        }
```

File: apps/backoffice-v2/src/common/components/organisms/Header/hooks/useNavbarLogic/useNavbarLogic.tsx [hunk 6]
```diff
@@ -57,6 +63,10 @@ export const useNavbarLogic = () => {
   const { pathname } = useLocation();
   const checkIsActiveFilterGroup = useCallback(
     (navItem: TRouteWithChildren) => {
+      if (!navItem.children) {
+        return navItem.href === pathname;
+      }
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/common/components/organisms/Header/hooks/useNavbarLogic/useNavbarLogic.tsx]: Revert please. This should never be true since a filter group always has children. The suggestion in the previous comment was general advice for the future.
